### PR TITLE
Update mergesort.js

### PR DESCRIPTION
**Pull request by bozidardj**
Oh, you fast JS :)

### CODE REVIEW

File: mergesort.js [hunk 1]
```diff
@@ -1,49 +1,6 @@
...
-      ret = ret.concat(m2);
-      m2 = [];
-    }
-    else{
-      break;
-    }
-  }
-
-  return ret;
+var mergesort = a => a.length < 2 ? a: merge(mergesort(a.splice(0, a.length/2)),mergesort(a))
```
Comment (geohot) [inline mergesort.js]: hmm, the second merge sort is being called on a. does a change?

Comment (bozidardj) [reply mergesort.js]: Hey, yes... that's the trick! Splice changes the original array, so we don't have to create two new arrays and waste precious resources :)

Comment (bozidardj) [reply mergesort.js]: Check this out => JS gets into the first place now! And it's the shortest one 🥇

Comment (jitwit) [reply mergesort.js]: still more chars than the J version (243 v 172) and the J's not golfed or squished into one line hah

Comment (bozidardj) [reply mergesort.js]: Yeah, J is the shorter one in terms of chars 👍  But JS is way more readable, anyone with enough programming experience can easily figure out what's going on

**Comment by geohot** [comment]
Unreadable one liner. No code golf!

**Pull request closed without merging**